test/tests/functional/pbs_stf.py

```diff
@@ -0,0 +1,761 @@
+        self.server.expect(JOB, attr, id=jid)
+
+        attr = {'Resource_List.walltime': (GE, '00:02:00')}
+        self.server.expect(JOB, attr, id=jid)
+
+        # The sleep below will leave less than 2 minutes window for jid
+        # after j2id is deleted.  The min_walltime of jid can't be
+        # satisfied and jid will stay in S state.
+        self.logger.info("Sleeping 65s to leave less than 2m before resv")
+        time.sleep(65)
+
+        a = {'queue': 'highp', 'Resource_List.select': '1:ncpus=1',
+             'Resource_List.walltime': '00:01:00'}
+        j = Job(TEST_USER, attrs=a)
+        j2id = self.server.submit(j)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2id)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+        time.sleep(5)
+        self.server.delete(j2id)
+        
+        t = int(time.time())
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.scheduler.log_match("Leaving Scheduling Cycle", starttime=t, 
+                                 max_attempts=5)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid)
+
+    def test_t_4_3_8(self):
+        """
+        Test shrink to fit by submitting a STF job and then creating a
+        reservation having ncpus=1 that overlaps with the job.  The
+        reservation is denied.
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        a = {'Resource_List.max_walltime': '02:00:00',
+             'Resource_List.min_walltime': '00:02:00'}
+        j = Job(TEST_USER, attrs=a)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+        now = int(time.time())
+        resv_start = now + 300
+        resv_dur = 7200
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.place': 'free',
+             'reserve_start': resv_start,
+             'reserve_duration': resv_dur
+             }
+        r = Reservation(TEST_USER, attrs=a)
+        rid = self.server.submit(r)
+        m = self.server.log_match(
+            "reservation deleted", starttime=self.server.ctime, max_attempts=10)
+        self.assertTrue(len(m) > 0)
+
+        self.server.delete(jid, wait=True)
+
+        a = {'Resource_List.select': '1:ncpus=1'}
+        j = Job(TEST_USER, attrs=a)
+        j2id = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=j2id)
+
+        resv_start = now + 300
+        resv_dur = 7200
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             'Resource_List.place': 'free',
+             'reserve_start': resv_start,
+             'reserve_duration': resv_dur
+             }
+        r = Reservation(TEST_USER, attrs=a)
+        self.server.submit(r)
+
+        m = self.server.log_match(
+            "reservation deleted", starttime=(now - 1), max_attempts=10)
```

Comment:
if starttime is now -1 then wouldn't it just match the previous denial of reservation. I think we nee to sleep for a second and then set now again

Reply:
I removed all the starttime parameters to log_match() and added the reservation id.  The log message is actually '<rid>;reservation deleted'